test/unit/knex.js

```diff
@@ -468,4 +468,30 @@ describe('knex', () => {
         });
     });
   });
+
+  describe('extend query builder', () => {
+    let connection;
+    beforeEach(() => {
+      connection = new sqlite3.Database(':memory:');
+    });
+
+    afterEach(() => {
+      connection.close();
+    });
+
+    it('should extend default queryBuilder', (done) => {
+      Knex.QueryBuilder.extend('customSelect', function(value) {
```

Comment:
This leaks extended querybuilder outside of thus test, I think. Would be good to delete custom metode in after() block.

Reply:
So you suggest to add remove method as well? cause with the current api, it always adds the method for the entire app.

Reply:
wouldn't simple delete `Knex.Querybuilder.prototype.customMethod` work? I don't think we should expose deletion method via API.

Reply:
It will work, I will update my PR soon

Reply:
Transaction is a different entity, not an original knex builder, but user would be justified to expect custom methods on it as well. It is highly likely we'd need to copy them over, but test would show that. Same with join builder, it's a different thing.